src/components/description_list/description_list.styles.ts

```diff
@@ -7,15 +7,21 @@
  */
 
 import { css } from '@emotion/react';
-import { logicalTextAlignCSS, euiMinBreakpoint } from '../../global_styling';
+import {
+  logicalTextAlignCSS,
+  euiMaxBreakpoint,
+  euiMinBreakpoint,
+} from '../../global_styling';
 import { UseEuiTheme } from '../../services';
 
 export const euiDescriptionListStyles = (euiThemeContext: UseEuiTheme) => {
-  // Flex display for column and responsive column
+  const TITLE_MAX_WIDTH = '380px';
```

Comment:
@kyrspl I think you chatted about this briefly in Slack, but can you clarify here how you came across this static `380px` width? And is there any way we can use EUI's theme tokens to calculate this end width instead of using a static number?

Reply:
At first, I thought about building a function that would calculate the relevant `max-width`. I came up with something like this:
```javascript
function calculateMaxWidth(containerWidth, fontSize, fontWeightFactor, maxCharactersPerLine) {
    // Estimate x-height and average character width based on font size and font weight factor
    const xHeightFactor = 0.546; // That's the closest I could approximate for Inter
    const averageCharacterWidth = fontSize * xHeightFactor * fontWeightFactor; // font weight factor is also an approximation

    // Return max width based on max characters per line and average character width
    return averageCharacterWidth * max_characters_per_line;
  }
```

Unfortunately, this is just an approximation. The end result heavily depends on our choice of max characters per line as well as x-height/font-weight factors. There could be other methods that provide a more accurate representation, but could be more resource intensive (e.g. CanvasRenderingContext2D: measureText() ). That said, as I wrote in the slack post, this is not an exact science. 

Even though I don't like using static definitions in the code, I thought it was a good start for fixing the existing (`column gap`) issue in this PR. 

Alternatively, we could create a utility function that calculates `max-width` and import it in the component.
One caveat though: This function would need to be tested against other components as well, if we want it to ensure its global utility. 

I'd suggest to go with a simple approach in this PR and research a more complete/all-encompassing one next.

What's your take?

Reply:
I agree it makes relatively little sense to over-engineer a function for this. However, I'd still prefer a compromise between a static px value and a complex utility.

Can we opt for a compromise of using the `ch` CSS property instead so the width at least dynamically adjusts to custom fonts/sizes/weights?

```suggestion
  const TITLE_MAX_WIDTH = '35ch';
```

Reply:
I looked into the `ch` unit in more detail: It computes the final width in `px`  according to the width of the `0` character of the nearest `font-size`. 

In practice, if we use the `ch` unit to compute the `max-width` of the first column, we will end up calculating that number according to a single `font-size`. However, the `description-list` can use different font sizes for the `title` and `description`, declared at the cell level. 

I thought we could do the following instead:

```css
/* grid display for column and responsive column */
display: grid;
grid-template-columns: minmax(auto,max-content) minmax(auto,max-content);
...
```

```css
/* define the max-width at the "title" and "description" level using the ch unit */
...
max-width: ${CONTENT_MAX_WIDTH}
```

Each cell will already have a `font-size` class (or inherit from the the nearest parent).

As a side note, I'm not particularly enamored with the `ch` unit because that metric heavily relies on the width of character `0`. This number can vary wildly from type to type, which could mess with the theme-ability of `EUI`. 

I could see us defining global content-width options which can be controlled at the theme level. Food for thought.

Reply:
Ahhh, I see your point about setting the width at the grid parent level rather than at the cell level. What you proposed looks fine to me as well. One alternative suggestion, if you'd still rather avoid `ch` but at least not use a static px value, is we could key off `euiTheme.size.base` instead:

```suggestion
  const TITLE_MAX_WIDTH = euiTheme.base * 24; // 384px
```

Again, this also isn't perfect and doesn't take custom fonts/font sizes into account, but honestly, I fairly strongly feel we're overthinking this for 99% of consumers. It's fairly rare that consumers customize the default theme in the first place, and those that do are willingly stepping into the territory of writing their own CSS if so.

The most flexible approach for the maximum number of consumers would likely be to create a new `titleMaxWidth` prop and allow consumers to override it on a case-by-case basis if necessary.

Reply:
That's a fair statement @cee-chen. I'm ok with using the `ch` unit, it's likely more accurate than the `px` value for these type of estimations. I've seen we also use it in `EuiText`. 

As an additional point - I think we could take it a step further and instead of defining a `max-width` we can use the logical property `max-inline-size` instead.